src/pykeen/training/training_loop.py

```diff
@@ -179,22 +203,39 @@ def train(
         # In some cases, e.g. using Optuna for HPO, the cuda cache from a previous run is not cleared
         torch.cuda.empty_cache()
 
-        result = self._train(
-            num_epochs=num_epochs,
-            batch_size=batch_size,
-            slice_size=slice_size,
-            label_smoothing=label_smoothing,
-            sampler=sampler,
-            continue_training=continue_training,
-            only_size_probing=only_size_probing,
-            use_tqdm=use_tqdm,
-            use_tqdm_batch=use_tqdm_batch,
-            tqdm_kwargs=tqdm_kwargs,
-            stopper=stopper,
-            result_tracker=result_tracker,
-            sub_batch_size=sub_batch_size,
-            num_workers=num_workers,
-        )
+        # If a checkpoint file is given we check whether it exists already and load it, if it does
+        if checkpoint_file:
+            if os.path.isfile(checkpoint_file):
+                stopper_dict = self._load_state(path=checkpoint_file)
+                # If the stopper dict has any keys, those are written back to the stopper
+                if stopper_dict:
+                    stopper._write_from_summary_dict(**stopper_dict)
+                continue_training = True
+            else:
+                logger.info(f"=> no checkpoint found at '{checkpoint_file}'. Creating a new file.")
+
+        # If the stopper loaded from the training loop checkpoint stopped the training, we return those results
+        if getattr(stopper, 'stopped', False):
```

Comment:
maybe `stopped` should just be set to false in the `NopStopper`

Reply:
I considered that as well. But this solution is more explicit to me, because it shows that only "real" stoppers can have stopped a training loop and thus have the attribute `stopped`.